internal/characters/barbara/abil.go

```diff
@@ -0,0 +1,197 @@
+package barbara
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+)
+
+// Standard attack function with seal handling
+func (c *char) Attack(p map[string]int) (int, int) {
+	f, a := c.ActionFrames(core.ActionAttack, p)
+
+	ai := core.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       fmt.Sprintf("Normal %v", c.NormalCounter),
+		AttackTag:  core.AttackTagNormal,
+		ICDTag:     core.ICDTagNormalAttack,
+		ICDGroup:   core.ICDGroupDefault,
+		Element:    core.Hydro,
+		Durability: 25,
+		Mult:       attack[c.NormalCounter][c.TalentLvlAttack()],
+	}
+	done := false
+	// Taken from Noelle code
+	cb := func(a core.AttackCB) {
+		if done { //why do we need this @srl
+			return
+		}
+		//check for healing
+		if c.Core.Status.Duration("barbskill") > 0 {
+			//heal target
+			heal := (prochpp[c.TalentLvlSkill()] + prochp[c.TalentLvlSkill()])
```

Comment:
i think you forgot to multiply this by her current hp? or is it max?

Reply:
maxhp... fixed